src/Cms/Pages.php

```diff
@@ -230,6 +230,15 @@ public function findById(string $id = null)
             return $page;
         }
 
+        // for secondary languages, try the full translated URI
+        if (
+            App::instance()->multilang() === true &&
+            App::instance()->language()->isDefault() === false &&
+            $page = $this->findBy('uri', $id)
+        ) {
+            return $page;
+        }
+
         $start = is_a($this->parent, 'Kirby\Cms\Page') === true ? $this->parent->id() : '';
         $page  = $this->findByIdRecursive($id, $start, App::instance()->multilang());
```

Comment:
```suggestion
        $start = is_a($this->parent, 'Kirby\Cms\Page') === true ? $this->parent->id() : '';
        $page  = $this->findByIdRecursive($id, $start, App::instance()->multilang());

        // for secondary languages, try the full translated URI
        if (
            App::instance()->multilang() === true &&
            App::instance()->language()->isDefault() === false &&
            $page = $this->findBy('uri', $id)
        ) {
            return $page;
        }
```

I would strongly suggest doing the lookup by `uri` _after_ letting the normal lookup occur by `id`. There is an existing oddity that `findById()` actually _prefers_ untranslated versions. Swapping the order should do a better job at keeping this behavior consistent.

See the end of the `testFindByIdWithSwappedSlugsTranslated()` test case lookups. https://github.com/getkirby/kirby/blob/27ae49e07a91323a05c8b87f950fa6ed27bb64ed/tests/Cms/Pages/PagesTest.php#L551-L653

Reply:
You are absolutely right, thanks for spotting! 👍